### PR TITLE
process precip to cumulative + color bins

### PR DESCRIPTION
**Pull request by lindsaycarr**
Processing for precip data.

### CODE REVIEW

File: viz.yaml [hunk 1]
```diff
@@ -1,333 +1,351 @@
...
-  publish-date: 2018-10-12
-  path: /how-gdp
-  analytics-id: 
-  description: >-
-    Explanation of how the Geo Data Portal works.
-  keywords: >-
-    USGS, VIZLAB, GDP
-  audience: General public
+id: how-gdp
+name: How GDP Works
```
Comment (wdwatkins) [inline viz.yaml]: If this is the actual title, we probably shouldn't use an acronym

File: viz.yaml [hunk 2]
```diff
@@ -1,333 +1,351 @@
...
+name: How GDP Works 
+date: 2017-09-19
+publish-date: 2018-10-12
+path: /how-gdp
+analytics-id: 
+description: >-
+  Explanation of how the Geo Data Portal works.
+keywords: >-
+  USGS, VIZLAB, GDP
+audience: General public
```
Comment (wdwatkins) [inline viz.yaml]: probably more internal?

File: scripts/process/process.utils.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+#' add a column to precip data based on the color bin  the precip
+#' value corresponds to
+#' 
+#' this function assumes column names `DateTime`, 
+#' `id` (location of precip summarization), and `precipVal`
+#' 
+bin_precip <- function(precip_df, breaks){
+  
+  precip_df <- dplyr::mutate(precip_df, cols = cut(precipVal, breaks = breaks, labels = FALSE))
```
Comment (aappling-usgs) [inline scripts/process/process.utils.R]: dplyr seems to me like a great candidate for the `@import` tag, despite the general encouragement to use `::`. you lose so much of the simplicity offered by dplyr when you take away pipes and have `::` in every line.

File: viz.yaml (hunk 2 above)
Comment (lindsaycarr) [reply viz.yaml]: yeah, probably more like "GDP current and future users"

File: viz.yaml (hunk 1 above)
Comment (lindsaycarr) [reply viz.yaml]: This is what ends up in the tab on Google chrome. Is that the "actual title" ?

File: scripts/process/process.utils.R (hunk 3 above)
Comment (lindsaycarr) [reply scripts/process/process.utils.R]: I agree -- using `%>%` is one of the main reasons I turn to dplyr

Comment (dblodgett-usgs) [reply scripts/process/process.utils.R]: I think this was at my suggestion. I agree that importing dplyr is the way to go, but I also REALLY like to have the package called out explicitly this code is so collaborative, that being super clear about what functions are what is super helpful. That said, dplyr is becoming such a core package, maybe the :: rule could be overlooked in this case.

Comment (lindsaycarr) [reply scripts/process/process.utils.R]: I don't mind `::` but I also want access to the pipes. What if we did both:

```
library(dplyr)
dplyr::mutate(df, new_col = 5*old_col) %>% 
    dplyr::select(new_col) 
```

Comment (aappling-usgs) [reply scripts/process/process.utils.R]: I appreciate the need to be clear about where functions come from. You could even clarify the origin of `%>%` with this line:
```
`%>%` <- magrittr::`%>%`
```
which would permit you to keep omitting the `library(dplyr)` call while also using pipes in the code that follows.

**Comment by lindsaycarr** [comment]
Merging since it's been approved

**Pull request merged**